src/writeData/components/PluginAddToExistingConfiguration/Configure.tsx

```diff
@@ -0,0 +1,143 @@
+    onSetTelegrafConfigID,
+    onSetTelegrafConfigName,
+    setIsValidConfiguration,
+    setPluginConfig,
+    telegrafConfigID,
+    telegrafs,
+  } = props
+
+  const selectedTelegraf = telegrafs?.find(
+    telegraf => telegraf.id === telegrafConfigID
+  )
+
+  if (!selectedTelegraf) {
+    setIsValidConfiguration(false)
+  }
+
+  const selectedName = selectedTelegraf
+    ? selectedTelegraf.name
+    : 'Select a configuration'
+
+  const handleSelectTelegrafConfiguration = (telegraf: Telegraf) => {
+    const targetBucketName = Array.isArray(telegraf?.metadata?.buckets)
+      ? telegraf.metadata.buckets[0]
+      : ''
+    const selectedBucket = buckets.find(
+      bucket => bucket.name === targetBucketName
+    )
+    if (selectedBucket) {
+      const {orgID, name, id} = selectedBucket
+      onSetBucketInfo(orgID, name, id)
+    }
+    onSetTelegrafConfigDescription(telegraf.description)
+    onSetTelegrafConfigID(telegraf.id)
+    onSetTelegrafConfigName(telegraf.name)
+    setPluginConfig('')
+    setIsValidConfiguration(true)
+  }
+
+  return (
+    <>
+      <div>Telegraf Agent Configuration</div>
+      <Dropdown
+        button={(active, onClick) => (
+          <Dropdown.Button
+            active={active}
+            onClick={onClick}
+            status={ComponentStatus.Default}
+          >
+            {selectedName}
+          </Dropdown.Button>
+        )}
+        menu={onCollapse => (
+          <Dropdown.Menu onCollapse={onCollapse}>
+            {telegrafs.map(telegraf => (
+              <Dropdown.Item
+                key={telegraf.id}
+                onClick={handleSelectTelegrafConfiguration}
+                selected={
+                  selectedTelegraf && telegraf.id === selectedTelegraf.id
+                }
+                value={telegraf}
+                testID={`telegraf-configuration--${telegraf.name}`}
+              >
+                {telegraf.name}
+              </Dropdown.Item>
+            ))}
+          </Dropdown.Menu>
+        )}
+        testID="plugin-add-to-configuration--select-telegraf"
+      />
+    </>
+  )
+}
+
+const mstp = (state: AppState) => {
+  const buckets = getAll<Bucket>(state, ResourceType.Buckets)
+  const {telegrafConfigID} = getDataLoaders(state)
+  const telegrafs = getAll<Telegraf>(state, ResourceType.Telegrafs)?.sort(
+    (firstConfig, secondConfig) => {
+      if (
+        firstConfig?.name.toLocaleLowerCase() >
+        secondConfig?.name.toLocaleLowerCase()
+      ) {
+        return 1
+      }
+      if (
+        firstConfig?.name.toLocaleLowerCase() <
+        secondConfig?.name.toLocaleLowerCase()
+      ) {
+        return -1
+      }
+      return 0
+    }
+  )
+
+  return {
+    buckets,
+    telegrafConfigID,
+    telegrafs,
+  }
+}
+
+const mdtp = {
+  onSetBucketInfo: setBucketInfo,
+  onSetTelegrafConfigDescription: setTelegrafConfigDescription,
+  onSetTelegrafConfigID: setTelegrafConfigID,
+  onSetTelegrafConfigName: setTelegrafConfigName,
```

Comment:
are you a fan of remapping the actions into this style?

I much prefer not renaming these, so something like:

```ts
const mdtp = {
  setBucketInfo,
  setTelegraf,
  setTelegrafConfigID,
  setTelegrafConfigName
}
```

i'd prefer to keep them named this way for 2 reasons:

1. `onAction` is usually the property that represents what to do in response to an event, eg `onClick`, `onSelect`, `onChange`. It's kind of awkward to have something like `onChange={onSetBucketInfo}`. In my opinion, it's much more natural to say `onChange={setBucketInfo}` - it reads like an english sentence
2. if you search the whole codebase for `setBucketInfo` (or worse, `setBucketInfo\(`) you won't be able to see `onSetBucketInfo`. You'll have to find where it's called in this function with a separate search.

Reply:
We have to rename them, or else there is a mirroring conflict. `mdtp` actually curries the functions in the object and creates new functions. Hence the renaming. If we keep the same name then there would be two functions named `setBucketInfo` - one of which is imported and one which is created by `mdtp`. It is very confusing to debug if there is ever an issue and the function is named the same thing even though the passed-in-prop version of that function has different arguments.

Reply:
>If we keep the same name then there would be two functions named setBucketInfo - one of which is imported and one which is created by `mdtp`

there are two functions named that, true. But it's only an issue if they're destructured out of the `props` object. Even then, it's not a collision issue, because the scope of the function component is more specific / deeper than the scope that the functions are imported into. (JavaScript is function scoped, so new functions create new variable scopes).

```ts
const name = 'Bucky'

const whoIsNumberOne = () => {
  const name = 'Timmy'
  return name
}

console.log(whoIsNumberOne()) // Timmy
```

I'd make the argument that keeping them within the props object (rather than destructuring) is actually the most documenting and easy to understand, especially in the context of a render method, or a function component.

```ts
const GiveMeTheTime: FC = (props) => {
  const computedValue = Date.now() * this.props.timeZone

  return (
    <div className={this.props.timeClassName}>
      {computedValue}
    </div>
  )
}
```

I think that this form is more self-documenting, because it is explicit about which values being used are from props, and which are computed (anything with `this.props` is from props, anything without it isn't).